pkg/pbq/store/wal/bootstrap.go

```diff
@@ -0,0 +1,198 @@
+package wal
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/gob"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/numaproj/numaflow/pkg/isb"
+	"github.com/numaproj/numaflow/pkg/pbq/partition"
+)
+
+var location *time.Location
+
+func init() {
+	var err error
+	location, err = time.LoadLocation("UTC")
+	if err != nil {
+		panic(fmt.Sprint("cannot load UTC", err))
+	}
+}
+
+// IsCorrupted checks whether the file is corrupt
+func (w *WAL) IsCorrupted() bool {
+	return w.corrupted
```

Comment:
I looked through the code and I do not think we are setting the corrupted flag everywhere.

Reply:
It is set at `(w *WAL) Read(size int64)` when checksum doesn't match.

Reply:
shouldn't it also be set if the header is bad, etc.?